packages/jest-test-result/src/types.ts

```diff
@@ -113,6 +113,11 @@ export type TestResult = {
   v8Coverage?: V8CoverageResult;
 };
 
+export type TestProgress = {
+  numRanTests: number;
+  numTotalTests: number;
```

Comment:
This is the minimal number of fields I needed to show progress. But it seems like reporters would want a lot more fields here, like whether any tests have failed, the assertion results, etc.

I am thinking about just passing a TestResult object here, but there isn't a field in there to show the number of total tests.

Reply:
>I am thinking about just passing a TestResult object here, but there isn't a field in there to show the number of total tests.

What about something like `type TestResultsProgress = TestResult & {...}`? I agree we should pass more info so a reporter could say how many failed (like what we want in the summary mentioned in #6616)